data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/SourceInfo.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.opensearch;
+
+public class SourceInfo {
+
+    private String osVersion;
+
+    private String dataSource;
+
+    private Boolean healthStatus = true;
+
+    public String getOsVersion() {
+        return osVersion;
+    }
+
+    public String getDataSource() {
+        return dataSource;
+    }
+
+    public void setDataSource(String dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    public void setOsVersion(String osVersion) {
+        this.osVersion = osVersion;
+    }
+
+    public Boolean getHealthStatus() {
```

Comment:
Additionally, Should this Cluster object make API's calls to actually fetch the Health of the cluster instead of relying on the provider to handle this?.

@mallikagogoi7 would you mine putting out a skeleton / boilerplate PR for this work? I think we can collaborate on class design here to improve the code structure. Thanks!